File: greedy/greedy_boss.py
"""
GREEDY – A COBIÇA ENCARNADA
Simulação de mecânica de buff e penalidade
Autor: Marcelo
"""

import random

class Greedy:
    def __init__(self):
        self.buff_gold = True
        self.buff_value = random.randint(300, 500)  # ouro concedido
        self.tentacle_hit = False

    def atacar(self):
        ataques = ["Moedas Douradas", "Pedras Preciosas", "Tentáculos Sombrios"]
        return random.choice(ataques)

    def aplicar_tentaculo(self, jogador):
        jogador["ouro_bonus"] = 0
        self.tentacle_hit = True
        return f"{jogador['nome']} perdeu o buff de ouro!"

    def derrotada(self, jogador):
        if not self.tentacle_hit:
            jogador["ouro_bonus"] += self.buff_value
            return f"{jogador['nome']} recebeu {self.buff_value} de ouro bônus!"
        else:
            return f"{jogador['nome']} não recebeu bônus devido ao tentáculo!"

# Exemplo de uso
if __name__ == "__main__":
    jogador = {"nome": "Herald1", "ouro_bonus": 0}
    boss = Greedy()
    print("Ataque de Greedy:", boss.atacar())
    print(boss.aplicar_tentaculo(jogador))
    print(boss.derrotada(jogador))

File: greedy/GreedyBoss.java
// GREEDY – A COBIÇA ENCARNADA
// Simulação de buff de ouro e penalidade por tentáculos
// Autor: Marcelo

import java.util.Random;

class Jogador {
    String nome;
    int ouroBonus;

    public Jogador(String nome) {
        this.nome = nome;
        this.ouroBonus = 0;
    }
}

public class Greedy {
    boolean buffGold = true;
    int buffValue;
    boolean tentacleHit = false;

    public Greedy() {
        this.buffValue = new Random().nextInt(201) + 300; // 300 a 500
    }

    public String atacar() {
        String[] ataques = {"Moedas Douradas", "Pedras Preciosas", "Tentáculos Sombrios"};
        return ataques[new Random().nextInt(ataques.length)];
    }

    public String aplicarTentaculo(Jogador j) {
        j.ouroBonus = 0;
        tentacleHit = true;
        return j.nome + " perdeu o buff de ouro!";
    }

    public String derrotada(Jogador j) {
        if (!tentacleHit) {
            j.ouroBonus += buffValue;
            return j.nome + " recebeu " + buffValue + " de ouro bônus!";
        } else {
            return j.nome + " não recebeu bônus devido ao tentáculo!";
        }
    }

    public static void main(String[] args) {
        Jogador j = new Jogador("Herald1");
        Greedy boss = new Greedy();

        System.out.println("Ataque de Greedy: " + boss.atacar());
        System.out.println(boss.aplicarTentaculo(j));
        System.out.println(boss.derrotada(j));
    }
}
